src/blockdata/witness.rs

```diff
@@ -198,17 +192,15 @@ impl<'a> Iterator for Iter<'a> {
     type Item = &'a [u8];
 
     fn next(&mut self) -> Option<Self::Item> {
-        let vec = &self.witness.content;
-        if self.cursor >= vec.len() {
-            None
-        } else {
-            let var = VarInt::consensus_decode(&vec[self.cursor..])
-                .expect("is granted witness.content contains varint because created only from internal methods");
-            let start = self.cursor + var.len();
-            let end = start + var.0 as usize;
-            self.cursor = end;
-            Some(&vec[start..end])
+        let varint = VarInt::consensus_decode(self.0.as_slice()).ok()?;
+        self.0.nth(varint.len() - 1)?; // VarInt::len returns at least 1
```

Comment:
I'd like to see `VarInt::consensus_decode_iter()` one day but good enough for this PR.